src/tools/pbs_sleep.c

```diff
@@ -70,13 +79,23 @@ main(int argc, char *argv[])
 		exit(1);
 	}
 
+	#ifndef WIN32
+	signal(SIGINT, handler);
+	signal(SIGHUP, handler);
+	signal(SIGTERM, handler);
+	#endif
+
 	/* if argv[1] is -1, loop with sleep 1 indefinitely */
-	if (strcmp(argv[1], "-1")) {
-		while(1)
+	if (strcmp(argv[1], "-1") == 0) {
+		while (getout == 0)
 			sleep(1);
+		return 0;
 	}
 
-	sleep(atoi(argv[1]));
+	while (i < atoi(argv[1]) && getout ==0) {
+		sleep(1);
+		i++;
+	}
```

Comment:
You could merge these two loops.  Have a local variable called something like 'forever'.  Set forever to 1 if pbs_sleep -1 is passed in.  Do a loop that checks forever || i < N.

Plus, use a for() loop.  You're iterating.  
Also, don't put atoi() in the loop header.  You'll convert it to an int N times.  Do the conversion once.